sycl/source/detail/graph_impl.cpp

```diff
@@ -226,6 +228,133 @@ bool graph_impl::clearQueues() {
   return AnyQueuesCleared;
 }
 
+/// Visits a node on the graph and it's successors recursively in a depth-first
+/// approach.
+/// @param Node The current node being visited.
+/// @param VisitedNodes A set of unique nodes which have already been visited.
+/// @param NodeStack Stack of nodes which are currently being visited on the
+/// current path through the graph.
+/// @param NodeFunc The function object to be run on each node. A return value
+/// of true indicates the search should be ended immediately and the function
+/// will return.
+/// @return True if the search should end immediately, false if not.
+bool visitNodeDepthFirst(
```

Comment:
can we put this in an anonymous namespace since it's a helper function.

Reply:
Done!